src/utils.rs

```diff
@@ -0,0 +1,94 @@
+//! Utility functions for the Sprocket command line tool.
+
+use std::path::PathBuf;
+use std::fs;
+use std::io::Read;
+
+use anyhow::{Result, Context};
+
+/// Determines if a file is a YAML file based on its extension.
+pub fn is_yaml_file(path: &PathBuf) -> bool {
```

Comment:
Haven't read through the serde docs, but can this be done more robustly by peeking file contents? File extensions aren't very reliable. I'd rather try to parse as YAML or JSON and return which is error free (and just fail if neither can be parsed)